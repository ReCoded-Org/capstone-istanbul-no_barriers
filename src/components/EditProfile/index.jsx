import React, { useState } from "react";
import { Modal, Form, Button } from "react-bootstrap";
import { useTranslation } from "react-i18next";

const EditProfile = ({ setShowEditModal, showEditModal, info, setInfo }) => {
  const { t } = useTranslation();
  const [input, setInput] = useState(info);

  const handleInputChange = (event, idInput) => {
    setInput((state) => ({ ...state, [idInput]: event }));
  };

  const handleSubmit = () => {
    setInfo(input);
    setShowEditModal(false);
  };

  return (
    <Modal
      size="lg"
      show={showEditModal}
      onHide={() => setShowEditModal(false)}
      aria-labelledby="editProfileModal"
      backdrop="static"
      keyboard={false}
    >
      <Modal.Header closeButton>
        <Modal.Title id="editProfileModal">
          {t("profile.editProfileModal.title")}
        </Modal.Title>
      </Modal.Header>
      <Modal.Body>
        <Form>
          <Form.Group>
            <Form.Label>{t("profile.editProfileModal.name")}</Form.Label>
            <Form.Control
              id="name"
              type="text"
              value={input.name}
              placeholder={t("profile.editProfileModal.namePlaceholder")}
              onChange={(e) => handleInputChange(e.target.value, e.target.id)}
            />
          </Form.Group>
          <Form.Group>
            <Form.Label>{t("profile.email")}</Form.Label>
            <Form.Control
              id="email"
              type="text"
              value={input.email}
              placeholder={t("profile.editProfileModal.emailPlaceholder")}
              onChange={(e) => handleInputChange(e.target.value, e.target.id)}
            />
          </Form.Group>
          <Form.Group>
            <Form.Label>{t("profile.phone")}</Form.Label>
            <Form.Control
              id="phoneNum"
              type="tel"
              value={input.phoneNum}
              placeholder={t("profile.editProfileModal.phonePlaceholder")}
              onChange={(e) => handleInputChange(e.target.value, e.target.id)}
            />
          </Form.Group>
          <Form.Group>
            <Form.File
              id="profilePic"
              value={input.profilePc}
              label={t("profile.editProfileModal.profileImage")}
              onChange={(e) => handleInputChange(e.target.value, e.target.id)}
            />
          </Form.Group>
          <Form.Group>
            <Form.Label>{t("profile.editProfileModal.bio")}</Form.Label>
            <Form.Control
              id="bio"
              as="textarea"
              rows={3}
              value={input.bio}
              placeholder={t("profile.editProfileModal.bioPlaceholder")}
              onChange={(e) => handleInputChange(e.target.value, e.target.id)}
            />
          </Form.Group>
        </Form>
      </Modal.Body>
      <Modal.Footer>
        <Button
          variant="outline-danger"
          onClick={() => setShowEditModal(false)}
        >
          {t("profile.editProfileModal.close")}
        </Button>
        <Button variant="success" onClick={handleSubmit}>
          {t("profile.editProfileModal.save")}
        </Button>
      </Modal.Footer>
    </Modal>
  );
};

export default EditProfile;
