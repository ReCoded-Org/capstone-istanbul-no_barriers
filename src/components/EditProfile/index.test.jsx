import React from "react";
import renderer from "react-test-renderer";
import EditProfile from "./index";
import profilePic from "../PersonalInformation/profile.jpg";

it("EditProfile renders correctly", () => {
  let info = {
    profilePicSource: profilePic,
    name: "IBO (ﾉ◕ヮ◕)ﾉ*:･ﾟ✧",
    email: "aish@aaa.com",
    phoneNum: "+900000",
    bio:
      "Vestibulum elementum felis magna, sit amet ullamcorper nunc viverra sed. Morbi at est id quam consequat aliquam. Morbi congue tellus quis nisl porta, in varius urna euismod. Aliquam at ligula diam. Fusce accumsan posuere venenatis. Nullam at lacus id diam iaculis finibus nec sed purus. Curabitur eget interdum turpis. Suspendisse pellentesque accumsan mattis. Nullam laoreet dictum metus nec vestibulum. Nulla ac libero a erat vehicula ullamcorper. Phasellus hendrerit mi at leo molestie, nec iaculis tellus suscipit. Vestibulum imperdiet bibendum urna sed condimentum. Sed eu libero nec ex pretium ullamcorper. Vivamus pellentesque leo ipsum, iaculis viverra nunc malesuada ut. Nullam volutpat, ipsum non mollis semper, tortor magna pretium mauris, quis tincidunt arcu lacus vitae diam. Etiam quis augue non libero tristique pellentesque sed quis urna. Proin in porttitor sapien. Class aptent taciti sociosqu ad litora torquent per conubia nostra, per inceptos himenaeos. Quisque porttitor commodo erat nec aliquam. Phasellus hendrerit, urna a congue iaculis, nulla leo posuere nulla, vel accumsan nunc nulla at magna.",
  };
  const setInfo = (newInfo) => (info = newInfo);
  let showEditModal = false;
  const setShowEditModal = (newState) => (showEditModal = newState);

  const tree = renderer
    .create(
      <EditProfile
        setShowEditModal={setShowEditModal}
        showEditModal={showEditModal}
        info={info}
        setInfo={setInfo}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
