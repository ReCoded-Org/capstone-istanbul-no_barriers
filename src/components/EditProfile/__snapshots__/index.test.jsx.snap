// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditProfile renders correctly 1`] = `null`;
